pkg/apis/core/validation/validation.go

```diff
@@ -4353,6 +4353,19 @@ func ValidateService(service *core.Service) field.ErrorList {
 
 	// external traffic fields
 	allErrs = append(allErrs, validateServiceExternalTrafficFieldsValue(service)...)
+
+	// internal traffic policy field
+	if utilfeature.DefaultFeatureGate.Enabled(features.ServiceInternalTrafficPolicy) {
```

Comment:
I think we can rmove the feature gate check here as long as we check for `nil` in the validation check. Since disabled feature gate means the field is always nil.